### PR TITLE
Add router for different models

### PR DESCRIPTION
**Pull request by AHReccese**
#### Reference Issues/PRs
#16 
#### What does this implement/fix? Explain your changes.
* add different models router(aka chain router)
* some refactorings upon v0.1 files to make it ready to add NN models support

### CODE REVIEW

File: pymilo/chains/linear_model_chain.py [hunk 1]
```diff
@@ -26,11 +26,14 @@ def is_linear_model(model):
     """
     Check if the input model is a sklearn's linear model.
 
-    :param model: given model
+    :param model: is a string name of a linear model or a sklearn object of it
     :type model: any object
     :return: check result as bool
     """
-    return type(model) in SKLEARN_LINEAR_MODEL_TABLE.values()
+    if(isinstance(model, str)):
```
Comment (sadrasabouri) [inline pymilo/chains/linear_model_chain.py:33]: `if isinstance(model, str):`

File: pymilo/chains/linear_model_chain.py [hunk 2]
```diff
@@ -26,11 +26,14 @@ def is_linear_model(model):
     """
     Check if the input model is a sklearn's linear model.
 
-    :param model: given model
+    :param model: is a string name of a linear model or a sklearn object of it
```
Comment (sadrasabouri) [inline pymilo/chains/linear_model_chain.py:29]: `:param model: name of a linear model or a sklearn object of it`

File: pymilo/chains/linear_model_chain.py [hunk 3]
```diff
@@ -158,8 +161,8 @@ def validate_input(model, command, is_inner_model):
     :return: None
     """
     if command == Command.SERIALIZE:
-        if get_sklearn_type(model) in SKLEARN_LINEAR_MODEL_TABLE.keys():
-            return
+        if is_linear_model(model):
+            return
```
Comment (sadrasabouri) [inline pymilo/chains/linear_model_chain.py:165]: Remove extra space at the end of this line.

File: CHANGELOG.md [hunk 4]
```diff
@@ -8,9 +8,14 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ### Added
 - `ndarray_to_list` method added to `GeneralDataStructureTransporter`
 - `list_to_ndarray` method added to `GeneralDataStructureTransporter` 
+- NN chain initialized
```
Comment (sadrasabouri) [inline CHANGELOG.md:11]: Change it to 
```
`neural_network_chain.py` chain
```
Try to be more objective in `CHANGELOG.md`.

File: pymilo/utils/test_pymilo.py [hunk 5]
```diff
@@ -25,9 +27,10 @@ def pymilo_test(model, model_name, test_data):
     """
     x_test, _ = test_data
 
+    export_model_path = "exported_linear_models" if is_linear_model(model) else "exported_neural_networks" if is_neural_network(model) else None
```
Comment (sadrasabouri) [inline pymilo/utils/test_pymilo.py:30]: Is there any more scalable approach (like dictionaries)?

File: pymilo/chains/neural_network_chain.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+def is_neural_network(model):
+    """
+    Check if the input model is a sklearn's neural network.
+
+    :param model: is a string name of a neural netowrk or a sklearn object of it
+    :type model: any object
+    :return: check result as bool
+    """
+    if(isinstance(model, str)):
```
Comment (sadrasabouri) [inline pymilo/chains/neural_network_chain.py:13]: `if isinstance(model, str):`

File: pymilo/chains/neural_network_chain.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+"""PyMilo chain for linear models."""
+from ..pymilo_param import SKLEARN_NEURAL_NETWORK_TABLE
+
+def is_neural_network(model):
+    """
+    Check if the input model is a sklearn's neural network.
+
+    :param model: is a string name of a neural netowrk or a sklearn object of it
```
Comment (sadrasabouri) [inline pymilo/chains/neural_network_chain.py:9]: `:param model: name of a neural network or a sklearn object of it`
+ note the typo in `network`

**Review by sadrasabouri — CHANGES_REQUESTED** [review]

File: pymilo/utils/test_pymilo.py (hunk 5 above)
Comment (AHReccese) [reply pymilo/utils/test_pymilo.py:30]: Sure, you're right.

File: CHANGELOG.md (hunk 4 above)
Comment (AHReccese) [reply CHANGELOG.md:11]: could you please explain it more? I mean being more objective ...

File: pymilo/chains/neural_network_chain.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    """
+    Return the transported (Serialized or Deserialized) model.
+
+    :param request: given neural network model to be transported
+    :type request: any object
+    :param command: command to specify whether the request should be serialized or deserialized
+    :type command: transporter.Command
+    :return: the transported request as a json string or sklearn neural network model
+    """
+    return
```
Comment (sadrasabouri) [inline pymilo/chains/neural_network_chain.py:30]: Do you want to complete this function in future PRs? If so I suggest moving it completely to the next PR.

**Review by sadrasabouri — APPROVED** [review]

File: pymilo/chains/neural_network_chain.py (hunk 8 above)
Comment (AHReccese) [reply pymilo/chains/neural_network_chain.py:30]: Yes exactly, but I can't since the main router works with it.

**Pull request merged**